kvbc/include/db_adapter.h

```diff
@@ -4,5 +4,51 @@
 
 #pragma once
 
+#include "kv_types.hpp"
+#include <optional>
+#include <utility>
+
+namespace concord::kvbc {
+
+class NotFoundException : public std::runtime_error {
+ public:
+  NotFoundException(const std::string& error) : std::runtime_error(("NotFoundException: " + error).c_str()) {}
+  const char* what() const noexcept override { return std::runtime_error::what(); }
+};
+
+class IDbAdapter {
+ public:
+  // Returns the added block ID.
+  virtual BlockId addBlockToBlockchain(const SetOfKeyValuePairs& updates) = 0;
+
+  // Adds a block from its raw representation and a block ID.
+  // Typically called by state transfer when a block is received and needs to be added.
+  virtual void addRawBlock(const RawBlock& rawBlock, const BlockId& blockId) = 0;
+
+  // Get block in its raw form
+  virtual RawBlock getRawBlock(const BlockId& blockId) const = 0;
+
+  // If found, return the actual version and the value.
+  virtual std::optional<std::pair<Value, BlockId>> getValue(const Key&, const BlockId& blockVersion) const = 0;
+
+  // Delete a block from the database
+  virtual void deleteBlock(const BlockId& blockId) = 0;
+
+  // Used to retrieve the latest block.
+  virtual BlockId getLastBlockchainBlockId() const = 0;
```

Comment:
I think naming it that way is confusing as `blockchain` implies part of the blockchain. In case of state transfer, as blocks arrive in reverse order, they are not yet part of the blockchain. This is valid for the merkle tree DB adapter, not necessarily valid for the direct KV adapter. Nevertheless, my suggestion is to have a distinction between blocks that are part of the blockchain and ones that are still in the process of ST, i.e. calling these methods as:

```
// equivalent to getLastRechableBlockId(), i.e. returning the ID of the last block that is part of the blockchain
virtual BlockId getLastBlockchainBlockId() const = 0;

// returns the latest known block ID (in case ST doesn't run in reverse order, it will return the received block with the highest ID)
virtual BlockId getLatestKnownBlockId() const = 0;
```

I am open to other naming suggestions too, but would prefer that it reflects the distinction mentioned above.

Reply:
I think lastReachableBlock reflects better its meaning. state transfer may "know" the real last block and in case of ST in progress they are not equal. Besides it's something in common termonology